esmf_utils/OutputInfo.F90

```diff
@@ -123,37 +130,47 @@ function get_vertical_dim_spec_names_bundle_info(info, rc) result(names)
       integer, optional, intent(out) :: rc
       integer :: status
       integer :: i
-      character(len=:), allocatable :: name
+      character(len=ESMF_MAXSTR) :: spec_name
 
+      spec_name = EMPTY_STRING
       names = StringVector()
       do i=1, size(info)
-         name = get_vertical_dim_spec_info(info(i), _RC)
-         if(find_index(names, name) == 0) call names%push_back(name)
+         spec_name = get_vertical_dim_spec_info(info(i), _RC)
+         _ASSERT(spec_name /= EMPTY_STRING, 'No vertical dim spec found.')
+         if(find_index(names, spec_name) == 0) call names%push_back(trim(spec_name))
       end do
       _RETURN(_SUCCESS)
 
    end function get_vertical_dim_spec_names_bundle_info
 
    function get_vertical_dim_spec_name_field(field, rc) result(spec_name)
-      character(len=:), allocatable :: spec_name
+      character(len=ESMF_MAXSTR) :: spec_name
       type(ESMF_Field), intent(in) :: field
       integer, optional, intent(out) :: rc
       integer :: status
       type(ESMF_Info) :: info
 
+      spec_name = EMPTY_STRING
       call ESMF_InfoGetFromHost(field, info, _RC)
       spec_name = get_vertical_dim_spec_info(info, _RC)
       _RETURN(_SUCCESS)
 
    end function get_vertical_dim_spec_name_field
 
    function get_vertical_dim_spec_info(info, rc) result(spec_name)
-      character(len=:), allocatable :: spec_name
+      character(len=ESMF_MAXSTR) :: spec_name
```

Comment:
```suggestion
      character(len=:), allocatable :: spec_name
```

Reply:
This private function is used for two purposes:
1) to obtain the name for its own sake
2) to obtain the name for determining the number of vertical levels

In the second case, the name is need to know if the vertical dim spec is named `VERTICAL_DIM_NONE`. We don't set the num_levels info when the name is `VERTICAL_DIM_NONE`. Maybe we should. In any case, right now that's how that is. Since it is a simple string comparison, I don't need a trim or an allocatable string.

In the first case, I trim the string at the higher level call for the bundle case. For the case of a single field, I don't trim it because of performance hit seems greater than the memory used for the string.

Reply:
For `VERTICAL_DIM_NONE` it should be an error to request number of levels.   Setting to 1 would be incorrect, but would lead to correct results is most (but not all) use cases.

Will discuss soon.